cmd/operator-sdk/add/api.go

```diff
@@ -22,38 +22,49 @@ import (
 
 	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/internal/genutil"
 	"github.com/operator-framework/operator-sdk/internal/scaffold"
+	"github.com/operator-framework/operator-sdk/internal/scaffold/ansible"
+	"github.com/operator-framework/operator-sdk/internal/scaffold/helm"
 	"github.com/operator-framework/operator-sdk/internal/scaffold/input"
 	"github.com/operator-framework/operator-sdk/internal/util/projutil"
+	"gopkg.in/yaml.v2"
 
 	log "github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
 )
 
 var (
-	apiVersion     string
-	kind           string
-	skipGeneration bool
+	apiVersion       string
+	kind             string
+	operatorType     string
+	skipGeneration   bool
+	helmChartRef     string
+	helmChartVersion string
+	helmChartRepo    string
 )
 
 func newAddAPICmd() *cobra.Command {
 	apiCmd := &cobra.Command{
 		Use:   "api",
 		Short: "Adds a new api definition under pkg/apis",
 		Long: `operator-sdk add api --kind=<kind> --api-version=<group/version> creates
-the api definition for a new custom resource under pkg/apis. This command
-must be run from the project root directory. If the api already exists at
-pkg/apis/<group>/<version> then the command will not overwrite and return
-an error.
+the api definition for a new custom resource under pkg/apis for Go-based operators, and 
+creates resource folder under /roles for Ansible-based operators.
+This command also creates resource folder under /helm-charts for Helm-based operators.
+For Ansible based operators this would add new Resource folder under roles/ directory.
```

Comment:
s/would// s/add/adds/

Reply:
How about something like:

> operator-sdk add api --kind=<kind> --api-version<group/version> creates an API definition for a new custom resource.
>
> For Go-based operators ...
> 
> For Ansible-based operators ...
>
> For Helm-based operators ...

That way, we can talk about what each of them does separately. And for helm, the required and optional flags are different, so it would probably be easier to describe all of these differences in separate paragraphs.